Add batch decomposition for torch.unsafe_chunk

guilhermeleobas opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #110862

This updates the docs as well to show `torch.unsafe_chunk`. Should the `unsafe_*` functions should not appear in the docs?

guilhermeleobas committed:
Add batch decomposition for torch.unsafe_chunk

[ghstack-poisoned]

kshitij12345 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -10425,6 +10425,14 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
+    OpInfo('unsafe_chunk',
+           dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16, torch.float16, torch.chalf),
+           sample_inputs_func=sample_inputs_chunk,
+           check_batched_forward_grad=False,
```
gradcheck uses legacy vmap implementation for batched forward and `unsafe_chunk` uses `unsafe_split` which doesn't have a batching rule for legacy vmap. (`unsafe_split` OpInfo also sets this to False)

kshitij12345 reviewed (approved):
LGTM! Thanks @guilhermeleobas 

cc: @zou3519 for binding approval.

zou3519 commented on docs/source/torch.rst:
```diff
@@ -133,6 +133,7 @@ Indexing, Slicing, Joining, Mutating Ops
     transpose
     unbind
     unsqueeze
+    unsafe_chunk
```
The unsafe_* APIs should not show up in the docs. We don't want users using them directly.

zou3519 reviewed (commented):
Please remove the mention from the docs, otherwise, this LGTM

guilhermeleobas committed:
Update on "Add batch decomposition for torch.unsafe_chunk"


This updates the docs as well to show `torch.unsafe_chunk`. Should the `unsafe_*` functions should not appear in the docs?


[ghstack-poisoned]

guilhermeleobas commented:
@pytorchbot merge

guilhermeleobas commented:
@pytorchbot merge

guilhermeleobas commented:
@pytorchbot merge

guilhermeleobas committed:
Update on "Add batch decomposition for torch.unsafe_chunk"


This updates the docs as well to show `torch.unsafe_chunk`. Should the `unsafe_*` functions should not appear in the docs?


[ghstack-poisoned]

guilhermeleobas committed:
Update on "Add batch decomposition for torch.unsafe_chunk"


This updates the docs as well to show `torch.unsafe_chunk`. Should the `unsafe_*` functions should not appear in the docs?


[ghstack-poisoned]

guilhermeleobas commented:
@pytorchbot merge
